In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

# Statistical Analysis

### Task 1 — T-test: BMI vs Diabetes (20 min)

1) Question: Is the difference in mean BMI between diabetic and non-diabetic patients statistically significant?

2) H₀: There is no significant difference in mean BMI between 
    diabetic and non-diabetic patients (μ_diabetic = μ_non-diabetic)

   H₁: There is a significant difference in mean BMI between 
    diabetic and non-diabetic patients (μ_diabetic ≠ μ_non-diabetic)



In [2]:
df = pd.read_csv("../data/processed/cleaned_data.csv")

In [3]:
bct = pd.crosstab(df["_BMI5"],df["Diabetes_binary"],normalize='index')
print(bct)

Diabetes_binary         0         1
_BMI5                              
1201.0           1.000000  0.000000
1202.0           1.000000  0.000000
1203.0           1.000000  0.000000
1205.0           1.000000  0.000000
1206.0           1.000000  0.000000
...                   ...       ...
5996.0           0.000000  1.000000
5997.0           0.500000  0.500000
5998.0           1.000000  0.000000
5999.0           0.000000  1.000000
6000.0           0.666667  0.333333

[4063 rows x 2 columns]


In [4]:
df.groupby("Diabetes_binary")["_BMI5"].count()

Diabetes_binary
0    1805498
1     314959
Name: _BMI5, dtype: int64

In [5]:
BMI_nondiabetic = df[df["Diabetes_binary"]==0]['_BMI5'].reset_index(drop=True)
BMI_diabetic = df[df["Diabetes_binary"]==1]['_BMI5'].reset_index(drop=True)

In [6]:
print(stats.ttest_ind(BMI_diabetic,BMI_nondiabetic))

TtestResult(statistic=np.float64(342.3376742355085), pvalue=np.float64(0.0), df=np.float64(2120455.0))


In [10]:
print(BMI_diabetic.mean()/100)
print(BMI_nondiabetic.mean()/100)

30.872705685501924
27.15175886652879


**Observation (T-test: BMI vs Diabetes)**
- p-value ≈ 0 (p < 0.001) — we reject H₀
- Conclusion: Mean BMI differs significantly between diabetic 
  (mean=30.87) and non-diabetic (mean=27.15) patients
- Clinical interpretation: Higher BMI is statistically associated 
  with diabetes status in this population

### Task 2 — Chi-Square: Exercise vs Diabetes

1) Question: Is there a significant association between exercise participation and diabetes status?

2) H0: There is no significant association between exercise participation and diabetes status.

   H1: There is a significant association between exercise participation and diabetes status.

In [7]:
ect = pd.crosstab(df["EXERANY2"],df["Diabetes_binary"],normalize=False)
print(stats.chi2_contingency(ect))

Chi2ContingencyResult(statistic=np.float64(25608.746370319685), pvalue=np.float64(0.0), dof=1, expected_freq=array([[ 448779.86885186,   78287.13114814],
       [1356718.13114814,  236671.86885186]]))


**Observation (Chi-Square: Exercise vs Diabetes)**
- p-value ≈ 0 (p < 0.001) — we reject H0
- Chi-square statistic : 25608.74
- Degress of freedom : 1
- Conclusion: Exercise participation and diabetes status are significantly associated
- Clinical interpretation: People who report no exercise show a different diabetes distribution than those who report exercise in this population

### Task 3 — Chi-Square: Depression vs Diabetes

1) Question: Is there a significant association between Depressive Disorder and diabetes status?

2) H0: There is no significant association between Depressive Disorder and diabetes status.

   H1: There is a significant association between Depressive Disorder and diabetes status.

In [8]:
act = pd.crosstab(df["ADDEPEV2"],df["Diabetes_binary"],normalize=False)
print(stats.chi2_contingency(act))

Chi2ContingencyResult(statistic=np.float64(13282.534214438241), pvalue=np.float64(0.0), dof=1, expected_freq=array([[1468322.39350951,  256140.60649049],
       [ 337175.60649049,   58818.39350951]]))


**Observation (Chi-Square: Depression vs Diabetes)**
- p-value ≈ 0 (p < 0.001) — we reject H0
- Chi-Square Statistic: 13282.53
- Degrees of Freedom - 1
- Conclusion: Depressive disorder and diabetes status are significantly associated
- Clinical interpretation: The diabetes distribution differs between those with and without a history of depressive disorder in this population

### Task 4 — Confidence Intervals for BMI

In [18]:
ci_diabetic = stats.t.interval(
    confidence=0.95,
    df = len(BMI_diabetic) -1,
    loc = BMI_diabetic.mean(),
    scale = stats.sem(BMI_diabetic)
)
print(ci_diabetic[0]/100, ci_diabetic[1]/100)

30.84935908751247 30.896052283491375


In [19]:
ci_non_diabetic = stats.t.interval (
    confidence = 0.95,
    df = len(BMI_nondiabetic) -1,
    loc = BMI_nondiabetic.mean(),
    scale = stats.sem(BMI_nondiabetic)
)
print(ci_non_diabetic[0]/100,ci_non_diabetic[1]/100)

27.143848008883086 27.159669724174492


**Observation (Task 4: BMI Confidence Intervals)**
- Diabetic CI (divided by 100): 30.8494 to 30.8961
- Non-diabetic CI (divided by 100): 27.1438 to 27.1597
- Overlap: No, these intervals do not overlap.
- Non-overlap implies the mean BMI is statistically different between the two groups at the 95% level.
- The intervals are extremely narrow because the sample size is very large, so the standard error $SE = SD / \sqrt{n}$ is tiny.